test/unit/taxonomix_test.rb

```diff
@@ -61,21 +61,42 @@ def setup
     assert_empty @dummy.locations
   end
 
-  test ".with_taxonomy_scope expands organizations and locations to actual values" do
-    org1 = FactoryGirl.create(:organization)
-    org2 = FactoryGirl.create(:organization)
-    org3 = FactoryGirl.create(:organization)
-    user = FactoryGirl.create(:user, :organizations => [org1, org2])
-    dummy_class = @dummy.class
+  describe '.with_taxonomy_scope' do
+    setup do
+      @org = FactoryGirl.create(:organization)
+      @loc = FactoryGirl.create(:location)
+    end
+
+    test 'expands organizations and locations to actual values' do
+      org2 = FactoryGirl.create(:organization)
+      org3 = FactoryGirl.create(:organization)
+      user = FactoryGirl.create(:user, :organizations => [@org, org2])
 
-    as_user(user) do
-      dummy_class.with_taxonomy_scope(nil, nil)
+      as_user(user) do
+        @dummy.class.with_taxonomy_scope(nil, nil)
+      end
+
+      assert_empty @dummy.class.which_location
+      assert_includes @dummy.class.which_organization, @org
+      assert_includes @dummy.class.which_organization, org2
+      refute_includes @dummy.class.which_organization, org3
+    end
+
+    test 'returns only objects in the given taxonomies' do
+      @dummy.name = 'foo'
+      @dummy.save
+      TaxonomixDummy.expects(:taxable_ids).returns([@dummy.id]).at_least_once
+      assert_equal 1, @dummy.class.with_taxonomy_scope(@loc, @org).count
+      assert_equal [@dummy], @dummy.class.with_taxonomy_scope(@loc, @org).to_a
```

Comment:
Maybe store the result of `@dummy.class.with_taxonomy_scope(@loc, @org)` so it's not called twice and it's more DRY.